### DISCUSSION TITLE
TeslaMate under Macos Monterey

### DISCUSSION THREAD

**Discussion by Ramses7420** [teslamate-org/teslamate#3669] [Q&A] [answered]
Hello everyone. I am French, so my English is not very good. I would like to know how to install and use Teslamate under MacOS Monterey. I followed the user manual, but when I connect to my address: http: //192.168.x.xx/4000, I have this error
![SCR-20240212-m62](https://github.com/teslamate-org/teslamate/assets/54696668/56f1e5f1-dadf-43b9-9e5f-81ce9059cc66)

Can you help me ?

**Comment by brianmay** [comment]
Can you copy and paste the logs here?

**Reply by Ramses7420** [reply]
Where can I find these logs?

**Reply by ktsenvolt** [reply]
docker-compose logs teslamate    or
docker-compose logs teslamate > logs.txt

**Comment by ktsenvolt** [comment]
I have followed the instructions below to MacOS, no issue at all.


https://youtu.be/SwapIMg66cs?si=83LfX0VA4DOPfVxL

**Comment by Ramses7420** [comment]
Here is my log
[com.docker.backend.log](https://github.com/teslamate-org/teslamate/files/14275514/com.docker.backend.log)

**Comment by brianmay** [comment]
Unfortunately, that is the Docker logs, not the teslamate logs. Which I think isn't going to help us.

**Reply by Ramses7420** [reply, marked as answer]
All it's OK for. Reinstall and work fine.

**Reply by Ramses7420** [reply]
Just a question. If I subscribe, will I be able to restore the Tesla data on my Mac or is it automatically recovered? Thank you for your answers

**Reply by brianmay** [reply]
What do you mean "If I subscribe"? Subscribe to what?

**Reply by Ramses7420** [reply]
There is neither a subscription to Teslamate?

**Reply by brianmay** [reply]
TeslaMate is free and open source software, and no payment is necessarily.  We don't have any sort of hosting service though, you will need to arrange a platform to run it on yourself.

No data is automatically restored, unless you can import it yourself. There are importers for TeslaFi and tesla-apiscraper, I have not used them, tried them, or even looked at them in anyway, so I cannot personally vouch for them.

**Reply by Ramses7420** [reply]
Sorry for my mistake

**Reply by brianmay** [reply]
No problem, glad to have clarified this for you.

**Reply by Ramses7420** [reply]
I allow myself to annoy you a little more. I operate Teslamate on my Mac. If for an unknown reason, my Mac were to go out and that I cannot restart it, in the event of absence, will the data of TelSamate be updated and accessible after restarting

**Reply by cwanja** [reply]
No, TeslaMate cannot backfill data from Tesla due to Tesla not providing this capability. If TeslaMate is not running, there will be a gap in data.

**Reply by Ramses7420** [reply]
In this case, is it possible to operate Teslamate on the cloud, and if so on which the cheapest server

**Reply by cwanja** [reply]
Yes and I believe there are tutorials for how to set that up for GCP and other VPS providers. Use your search engine of choice to locate those. 

Just be aware, if your server or cloud provider is down or has an outage, the same concept applies - gap in the data.

**Reply by Ramses7420** [reply]
Many thanks for your help

**Reply by ktsenvolt** [reply]
You may want to try this: https://pimylifeup.com/raspberry-pi-teslamate/
I have personally used this since 2021, with no issues at all.
